src/Nethermind/Nethermind.Merge.Plugin/Handlers/V1/ForkchoiceUpdatedV1Handler.cs

```diff
@@ -352,8 +352,8 @@ private BlockInfo[] GetBeaconChainBranch(Block newHeadBlock, BlockInfo newHeadBl
                 }
                 BlockInfo predecessorInfo = _blockTree.GetInfo(predecessor.Number, predecessor.GetOrCalculateHash()).Info;
                 predecessorInfo.BlockNumber = predecessor.Number;
-                if (predecessorInfo.IsBeaconMainChain) break;
-                if (_logger.IsInfo) _logger.Info($"Reorged to beacon block ({predecessorInfo.BlockNumber}) {predecessorInfo.BlockHash} or cache rebuilt after restart");
+                if (predecessorInfo.IsBeaconMainChain || !predecessorInfo.IsBeaconInfo) break;
```

Comment:
Should we also have some depth limit?
Should we also check if something is just MainChain?

Reply:
in theory depth limit might be useful only in case of our bug, but I'm wondering what we can set here. Do you think we should break after, let's say 2000 blocks? 

If something is on main chain we should break here: !predecessorInfo.IsBeaconInfo